testing/time_test.ts

```diff
@@ -632,3 +634,88 @@ Deno.test("Date from FakeTime is structured cloneable", () => {
   assert(date instanceof Date);
   assert(cloned instanceof Date_);
 });
+
+Deno.test("Faked timer functions throws when called after FakeTime is restored", () => {
+  let fakeSetTimeout: typeof setTimeout;
+  let fakeClearTimeout: typeof clearTimeout;
+  let fakeSetInterval: typeof setInterval;
+  let fakeClearInterval: typeof clearInterval;
+  {
+    using _time: FakeTime = new FakeTime();
+    fakeSetTimeout = setTimeout;
+    fakeClearTimeout = clearTimeout;
+    fakeSetInterval = setInterval;
+    fakeClearInterval = clearInterval;
+  }
+  assertThrows(() => fakeSetTimeout(() => {}, 0), TimeError, "no fake time");
+  assertThrows(() => fakeClearTimeout(0), TimeError, "no fake time");
+  assertThrows(() => fakeSetInterval(() => {}, 0), TimeError, "no fake time");
+  assertThrows(() => fakeClearInterval(0), TimeError, "no fake time");
+});
+
+Deno.test("Faked Date.now returns real time after FakeTime is restored", () => {
+  let fakeDateNow: typeof Date.now;
+  {
+    using _time: FakeTime = new FakeTime();
+    fakeDateNow = Date.now;
+  }
+  assertAlmostEquals(Date.now(), fakeDateNow());
+});
+
+Deno.test("FakeTime can be called multiple times (later call is effective)", () => {
+  const now = Date.now();
+  try {
+    const _time1 = new FakeTime(1000);
+    const _time2 = new FakeTime(2000);
+    assertEquals(Date.now(), 2000);
+  } finally {
+    FakeTime.restore();
+  }
+  assert(Date.now() >= now);
+});
+
+Deno.test("FakeTime can be constructed with number, Date, or string", () => {
+  try {
+    const _time1 = new FakeTime(1000);
+    assertEquals(Date.now(), 1000);
+    const _time2 = new FakeTime(new Date(2000));
+    assertEquals(Date.now(), 2000);
+    const _time3 = new FakeTime("Thu Jan 01 1970 00:00:03 GMT+0000");
+    assertEquals(Date.now(), 3000);
+  } finally {
+    FakeTime.restore();
+  }
+});
+
+Deno.test("FakeTime throws when NaN is provided", () => {
+  assertThrows(() => new FakeTime(NaN), TimeError, "invalid start");
+});
+
+Deno.test("FakeTime.restore() throws when the time is already restored", () => {
+  const _time = new FakeTime();
+  FakeTime.restore();
+  assertThrows(() => FakeTime.restore(), TimeError, "time already restored");
+});
+
+Deno.test("time.restore() throws when the time is already restored", () => {
+  const time = new FakeTime();
+  time.restore();
+  assertThrows(() => time.restore(), TimeError, "time already restored");
+});
```

Comment:
This won't be needed once #5125 is merged.

Reply:
I think this case now makes sense as now we prohibit the double creations of FakeTime #5130